### PR TITLE
Platform.js

### PR DESCRIPTION
**Pull request by SeijiEmery**
Experimental script that spawns a procedural platform that you can stand on and which follows your avatar around. 

This is a neat demo that showcases the capabilities of our entity system (in terms of scale and coordinating many entities via javascript), and it allows you to make realtime changes to the platform as others look on (note: this can become quite bandwidth intensive!).

This script has  several known issues, and should probably not be merged or made public until these have been resolved:
– High volume entity edits can crash the client (https://app.asana.com/0/26225263936266/47157911701038). This is mostly a platform.js issue (since it can create thousands of entities), but it has been observed infrequently in other scripts (that also edit/create/destroy entities).
– 2d overlays can bug out and not get created after a script that uses them gets reloaded after a certain number of times (https://app.asana.com/0/26225263936266/47157911701042). Once again, likely caused by platform.js using a large number of overlays for its UI
– UI lag (where the ui starts spawning in the top left corner of the screen and it visibly takes several seconds to assemble itself). Non-critical, but this is a uiwidgets issue and will be fixed in a future PR.

You can try out the script here: 
https://raw.githubusercontent.com/SeijiEmery/hifi/platform-js/examples/example/entities/platform.js

### CODE REVIEW

**Comment by SeijiEmery** [comment]
Needs review.
Will clean up / rewrite the description as soon as this is OK to merge.

File: examples/example/entities/platform.js [hunk 1]
```diff
@@ -0,0 +1,1210 @@
...
+
+			Controller.mousePressEvent.connect(UI.handleMousePress);
+			Controller.mouseMoveEvent.connect(UI.handleMouseMove);
+			Controller.mouseReleaseEvent.connect(UI.handleMouseRelease);
+		}
+	}
+	Script.update.connect(this.startup);
+})();
+
+})();
```
Comment (AndrewMeadows) [inline examples/example/entities/platform.js]: Tab indentation.  Really?

Some people will tell you to use 2-spaces per indentation level on JS, but don't listen to them.  4-space is the way to go.

Comment (ZappoMan) [inline examples/example/entities/platform.js]: I agree with Andrew. This needs to be fixed. The coding standard is clear. All indents should be 4 spaces.

Comment (PhilipRosedale) [inline examples/example/entities/platform.js]: did this get fixed, @SeijiEmery

Comment (PhilipRosedale) [inline examples/example/entities/platform.js]: @SeijiEmery Did you fix these problems?

**Comment by PhilipRosedale** [comment]
Closing until you have time to work on this.  :(

**Pull request closed without merging**